### PR TITLE
Product shop previews

### LINKED ISSUE #30734: Add a preview of all shops with their associated data
### Product Preview

WHEN the product is associated to more than one store
THEN A dropdown is displayed next to the list of store. 

WHEN the dropdown is pressed 
THEN
It add to the product list all the product in each store in a preview panel. The inital row displays '-' for all the column

image

Preview should load with Ajax

### PR DESCRIPTION
**Pull request by jolelievre**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Partial PR for product list V2 in multishop, handle the toggle of product's shops preview, all the associated shops are displayed line by line. This is not the final PR for this feature as the action have not been handled yet.
| Type?             | new feature
| Category?         | BO
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Fixes #30734
| Related PRs       | ~
| How to test?      | ~
| Possible impacts? | ~

**Screenshot**

Capture d’écran 2022-12-27 à 11 51 14

### CODE REVIEW

**Review by SharakPL — DISMISSED** [review]

**Review by 0x346e3730 — DISMISSED** [review]
lgtm

File: src/Adapter/Product/Grid/Data/Factory/ProductShopsGridDataFactoryDecorator.php [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+use PrestaShop\PrestaShop\Core\Grid\Search\ShopSearchCriteriaInterface;
+use Shop;
+
+/**
+ * Decorates product grid data, but instead of adding details of all shops for the associated column, we prepare detail
+ * for single shop on each row.
+ */
+class ProductShopsGridDataFactoryDecorator extends ProductGridDataFactoryDecorator
+{
+    protected function applyShopModifications(array $products, ShopSearchCriteriaInterface $searchCriteria): array
```
Comment (zuk3975) [inline src/Adapter/Product/Grid/Data/Factory/ProductShopsGridDataFactoryDecorator.php:47]: arrays should probably be defined un docblock

File: src/Core/Grid/Definition/Factory/ProductGridDefinitionFactory.php [hunk 2]
```diff
@@ -269,6 +262,19 @@ protected function getColumns()
         return $columns;
     }
 
+    protected function getEditColumnAttributes(): array
```
Comment (zuk3975) [inline src/Core/Grid/Definition/Factory/ProductGridDefinitionFactory.php:268]: return type could be defined with array types as even the array keys are static

File: src/Core/Grid/Definition/Factory/ProductShopsGridDefinitionFactory.php [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+     */
+    protected function getEditColumnAttributes(): array
+    {
+        return [];
+    }
+
+    /**
+     * We don't perform any filtering on the shop, so no need to define them.
+     */
+    protected function getFilters()
```
Comment (zuk3975) [inline src/Core/Grid/Definition/Factory/ProductShopsGridDefinitionFactory.php:136]: i know its unused, but still needs a return type or inheritDoc I guess

**Review by zuk3975 — DISMISSED** [review]
just some missing docblocks/return types

**Review by zuk3975 — DISMISSED** [review]

File: src/Core/Grid/Definition/Factory/ProductShopsGridDefinitionFactory.php [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+                    'id_product'
+                )
+            )*/
+        ;
+
+        return $rowActions;
+    }
+
+    /**
+     * Edit attributes are only used to hanlde the shop selection modal, not needed in a shop row.
```
Comment (SharakPL) [inline src/Core/Grid/Definition/Factory/ProductShopsGridDefinitionFactory.php]:
```suggestion
     * Edit attributes are only used to handle the shop selection modal, not needed in a shop row.
```

**Review by zuk3975 — APPROVED** [review]

**Review by lartist — APPROVED** [review]

**Comment by jolelievre** [comment]
This PR is only a partial feature so it is not tested manually yet

**Pull request merged**